Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -638,13 +638,14 @@ static inline bool shouldAddContentKeyRecipients()
 
     ALWAYS_LOG(LOGIDENTIFIER);
 
+    auto shouldAddParser = shouldAddContentKeyRecipients() ? CDMSessionMediaSourceAVFObjC::ShouldAddParser::Yes : CDMSessionMediaSourceAVFObjC::ShouldAddParser::No;
     if (m_session)
-        m_session->removeSourceBuffer(this);
+        m_session->removeSourceBuffer(this, shouldAddParser);
```

Comment:
Wouldn't it be best to have the SourceBuffer do the work of adding the parser instead? this makes the CDMSession immediately calls back into the SourceBuffer fetch what should be a private member of the SourceBuffer

The SourceBuffer has the information on if its parser should be added or not.